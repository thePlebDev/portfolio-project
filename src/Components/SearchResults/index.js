import React,{useEffect,useState} from 'react'
import styled from 'styled-components'

const Container = styled.div`

  border:1px solid green;
  margin-top:5px
`
const Articles = styled.div`
  broder:1px solid blue;
  background-color:#FDFDFD;

`



const SearchResults =()=>{
  const [state,setState] = useState()
  useEffect(()=>{
    let cachedHits = sessionStorage.getItem('posts')
    setState(JSON.parse(cachedHits))
  },[])

  return(
    <Container>
        <Articles>
          Articles
          <div>
            {
              state
                ?
                state.map((item)=>{
                  return <div key={item.key}>
                    <div>
                      {item.title}
                    </div>
                    <div>
                      {item.description}
                    </div>
                    <div>
                      {item.filters}
                    </div>
                  </div>
                })
                :
                ''
            }
          </div>
        </Articles>
    </Container>
  )
}

export default SearchResults
