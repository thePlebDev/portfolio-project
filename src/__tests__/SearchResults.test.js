import React from 'react'

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SearchResults from '../Components/SearchResults';




describe('testing the SearchResults component',()=>{
  it('should render the snapshot',()=>{
    const tree = renderer.create(<SearchResults />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render the important stuff with default props',()=>{
    const wrapper = shallow(<SearchResults />)
    expect(wrapper.find('[data-testid="searchResultsFalse"]').length).toBe(1)
    expect(wrapper.find('[data-testid="searchResultsTrue"]').length).toBe(0)
  })
  it('should render the important stuff with custom props',()=>{
    const searchResults =[{name:'another one',_id:3}]
    const wrapper = shallow(<SearchResults searchResults={searchResults}  />)
    expect(wrapper.find('[data-testid="searchResultsTrue"]').length).toBe(1)
    expect(wrapper.find('[data-testid="searchResultsFalse"]').length).toBe(0)
  })
})
