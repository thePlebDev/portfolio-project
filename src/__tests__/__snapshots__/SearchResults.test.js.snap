// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the SearchResults component should render the snapshot 1`] = `
<div
  className="sc-eCApGN bQOiQb"
>
  <div
    className="sc-gKAblj elFhul"
  >
    Articles
  </div>
  <div
    className="sc-jSFkmK eiJvjg"
  >
    <div
      data-testid="searchResultsFalse"
      style={
        Object {
          "fontSize": "30px",
          "margin": "10px",
          "textAlign": "center",
        }
      }
    >
      No articles found
    </div>
  </div>
</div>
`;
